test/Test/Database/LSMTree/Internal/Merge.hs

```diff
@@ -112,6 +113,27 @@ prop_MergeDistributes fs hbio level stepSize (SmallList rds) =
     vals = concatMap (bifoldMap pure mempty . snd) kops
     isLarge = uncurry entryWouldFitInPage
 
+    getRunContent run@(DeRef Run.Run {
+                         Run.runFilter,
+                         Run.runIndex,
+                         Run.runKOpsFile,
+                         Run.runBlobFile
+                       }) = do
+      runSize         <- evaluate (Run.size run)
```

Comment:
Why `evaluate`?

Reply:
To ensure the `DeRef` is not delayed. If it's delayed past the `releaseRef` it's a use-after-free bug.

Reply:
I thought that since we're matching on `Run.Run`, that the assertion will have run by then

Reply:
The `Run.size` has its own use of `ReRef` which does a use-after-free check.